plugins/vbe/fit.h

```diff
@@ -0,0 +1,278 @@
+ * struct fit_info - Information about a Flat Image Tree being processed
+ *
+ * @blob: Pointer to FIT data (format is device tree binary / dtb)
+ * @size: Size of FIT data in bytes
+ */
+struct fit_info {
+	const char *blob;
+	int size;
+};
+
+/**
+ * fit_open() - Open a FIT ready for use
+ *
+ * The FIT must be entirely within in the buffer, but it may have external data
+ * in which case this appears after the FIT.
+ *
+ * @fit: Place to put info about the FIT
+ * @buf: Buffer containing the FIT
+ * @size: Size of the buffer
+ * Returns: 0 if OK, -ve fit_err_t on error
+ */
+int fit_open(struct fit_info *fit, const void *buf, size_t size);
+
+/**
+ * fit_close() - Shut down a FIT after use
+ *
+ * This frees any memory in use
+ *
+ * @fit: FIT to shut down
+ */
+void fit_close(struct fit_info *fit);
+
+/**
+ * fit_strerror() - Look up a FIT error number
+ *
+ * Since all errors are negative, this should be a negative number. If not, then
+ * a placeholder string is returned
+ *
+ * @err: Error number (-ve value)
+ * Returns: string corresponding to that error
+ */
+const char *fit_strerror(int err);
+
+/**
+ * fit_first_cfg() - Find the first configuration in the FIT
+ *
+ * @fit: FIT to check
+ * Returns: offset of first configuration, or -EFIT_NOT_FOUND if not found
+ */
+int fit_first_cfg(struct fit_info *fit);
+
+/**
+ * fit_next_cfg() - Find the next configuration in the FIT
+ *
+ * @fit: FIT to check
+ * @prev_cfg: Offset of the previous configuration
+ * Returns: offset of next configuration, or -EFIT_NOT_FOUND if not found
+ */
+int fit_next_cfg(struct fit_info *fit, int preb_cfg);
+
+/**
+ * fit_cfg_name() - Get the name of a configuration
+ *
+ * @fit: FIT to check
+ * @cfg: Offset of configuration node to check
+ * @return name of configuration, or NULL if @cfg is invalid
+ */
+const char *fit_cfg_name(struct fit_info *fit, int cfg);
+
+/**
+ * fit_cfg_compat_item() - Get the name of one of a configs's compat strings
+ *
+ * The config has a list of compatible strings, indexed from 0. This function
+ * returns am indexed string
+ *
+ * @fit: FIT to check
+ * @cfg: Offset of configuration node to check
+ * @index: Index of compatible string (0 for first, 1 for next...)
+ * Returns: Configuration's compatible string with that index, or NULL if none
+ */
+const char *fit_cfg_compat_item(struct fit_info *fit, int cfg, int index);
+
+/**
+ * fit_cfg_image_count() - Get the number of images in a configuration
+ *
+ * This returns the number of images in a particular configuration-node
+ * property. For example, for:
+ *
+ *	firmware = "u-boot", "op-tee";
+ *
+ * this would return 2, since there are two images mentioned.
+ *
+ * @fit: FIT to check
+ * @cfg: Offset of configuration node to check
+ * @prop_name: Name of property to look up
+ * Returns: Number of images in the configuration, or -ve if the offset is
+ * invalid or the proprerty is not found
+ */
+int fit_cfg_img_count(struct fit_info *fit, int cfg, const char *prop_name);
```

Comment:
Normally the docs go in the .c for glib, not the .h

Reply:
See above re style